pkg/operator/ceph/object/objectstore.go

```diff
@@ -757,15 +759,48 @@ func enableRGWDashboard(context *Context) error {
 	if err != nil || errCode != 0 {
 		return errors.Wrapf(err, "failed to create user %q", DashboardUser)
 	}
-	args := []string{"dashboard", "set-rgw-api-access-key", *u.AccessKey}
-	cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
-	_, err = cephCmd.Run()
-	if err != nil {
-		return errors.Wrapf(err, "failed to set user %q accesskey", DashboardUser)
+
+	cephVersion := mgr.FileBasedPasswordSupported(context.clusterInfo)
+	var args []string
+	var secretFile *os.File
+	// for latest Ceph versions
+	if cephVersion {
+		file, err := mgr.CreateTempPassFile(*u.AccessKey)
+		if err != nil {
+			return err
+		}
+
+		args = []string{"dashboard", "set-rgw-api-access-key", "-i", file.Name()}
+		cephCmd := ceph.NewCephCommand(context.Context, context.clusterInfo, args)
+		_, err = cephCmd.Run()
+		if err != nil {
+			return errors.Wrapf(err, "failed to set user %q accesskey", DashboardUser)
+		}
+		defer func() {
```

Comment:
This defer should be immediately after the file is created, for example on line 772

Reply:
I tried same but got error similar to `can't open file.name()failed to set user... accesskey`

Reply:
Maybe that was from the other one because of the goroutine? I don't see why this defer would cause a problem. Anyway, if the command is already complete here, you could just delete the file directly and no need for a defer. Since it's just a temp file that will go away with the container restart, it's really not that critical to be deleted anyway.